### PR TITLE
Remove flaky test by avoiding reuse of name for replicaset

### LINKED ISSUE #69376: ReplicaSetController can miss handling the deletion of a ReplicaSet
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:

@ncdc's description hoisted from https://github.com/kubernetes/kubernetes/issues/69376#issuecomment-427453571:

I was about to file this. I spent yesterday triaging it and have determined the root cause. The issue is that the `ReplicaSetController` can "miss" handling the deletion of a `ReplicaSet` if things happen quickly enough. Here's the flow:

1. Client creates rs
1. `ReplicaSetController` sees new rs, starts working on creating pods
1. Client deletes rs
1. `ReplicaSetController`'s `rsInformer` see the rs deletion and calls `rsc.enqueueReplicaSet`, which adds the namespace/name of the rs to the work queue
1. Client recreates rs with the exact same name as before
1. `ReplicaSetController`'s `rsInformer` see the rs addition and calls `rsc.enqueueReplicaSet`, which adds the namespace/name of the rs to the work queue (again)
1. `ReplicaSetController`'s sync handler processes the entry from the queue
1. Because the rs was recreated with the same name, when `syncReplicaSet` calls `rsLister` to get the rs, it's there (it's the 2nd one)

This is a timing issue. The `ReplicaSetController` doesn't check the rs's UID, and if the order of operations is "just right", the controller's sync handler won't "see" the deletion, so it never calls `rsc.expectations.DeleteExpectations(key)` to reflect that the rs was deleted.

/sig apps

Original description follows:

In #69344 I had to re-run the integration test 4 times to get it to work.

Log outputs:

* [attempt 1](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/69344/pull-kubernetes-integration/29684/)
* [attempt 2](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/69344/pull-kubernetes-integration/29685/)
* [attempt 3](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/69344/pull-kubernetes-integration/29688/)
* [attempt 4 (successful)](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/69344/pull-kubernetes-integration/29691/)

### PR DESCRIPTION
**Pull request by mortent**
**What this PR does / why we need it**: As reported in #69376, run_rs_tests is flaky and the underlying cause is a problem with ReplicaSets when names are reused. The run_rs_tests actually tests kubectl, so this PR removes the flakiness by avoiding this issue with ReplicaSets. It should fix the flakiness and the issue in ReplicaSet can be fixed separately.

It complicates the test a bit with the use of sed, but it seemed better than using sleep or adding additional test data.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #69376

**Special notes for your reviewer**:
@kow3ns 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/cmd/apps.sh [hunk 1]
```diff
@@ -510,11 +510,12 @@ run_rs_tests() {
   # Pre-condition: no replica set exists
   kube::test::get_object_assert rs "{{range.items}}{{$id_field}}:{{end}}" ''
   # Command
-  kubectl create -f hack/testdata/frontend-replicaset.yaml "${kube_flags[@]}"
+  local replicaset_name="frontend-no-cascade"
+  sed -r 's/^(\s*)(name\s*:\s*frontend\s*$)/\1name: '"${replicaset_name}"'/' hack/testdata/frontend-replicaset.yaml | kubectl create "${kube_flags[@]}" -f -
```
Comment (lavalamp) [inline test/cmd/apps.sh]: Add a todo to revert this after the bug is fixed?

Comment (mortent) [reply test/cmd/apps.sh]: Added todo

**Comment by spiffxp** [comment]
/approve
I'm approving this in the hopes that it will de-flake https://github.com/kubernetes/kubernetes/issues/69376

**Pull request merged**